### PR TITLE
CU-34awu9w/platform-repo-cleanup

### PR DESCRIPTION
**Pull request by RyanJMah**
This is a **SUPER** big PR because I regenerated the libraries (and most of the unnecessary library files got deleted). It is probably easier to just not look at the diffs and just read the main files of interest, which are:
1. ./makefile
2. ./STM32G473CBTx/makefile

Things that were done.
1. Created a scalable (imo) directory structure and build system for this project
2. Made the makefile create a static library of all source files in the `platform` directory, to be linked with future application code

PS: Adding Ethan as a reviewer since I remember (I think) he was saying he wanted to learn more about build systems

### CODE REVIEW

**Comment by Nushaab-Syed** [comment]
Task linked: [CU-34awu9w Platform Repo Cleanup](https://app.clickup.com/t/34awu9w)

**Comment by jennchenn** [comment]
Task linked: [CU-2zvpc9y Firmware Onboarding](https://app.clickup.com/t/2zvpc9y)

File: makefile [hunk 1]
```diff
@@ -1,220 +1,19 @@
...
-######################################
-# target
-######################################
-TARGET = main
+THIS_DIR := $(shell readlink -f .)
+BUILD_DIR = $(THIS_DIR)/build
 
-DEVICE_DIRNAME= STM32G473CBTx
+all:
+	cd $(DEV) && make BUILD_DIR=$(BUILD_DIR)
```
Comment (Quinn1876) [inline makefile:5]: is DEV defined somewhere?

Comment (RyanJMah) [reply makefile:5]: I intended it to be passed as a CLI argument.

Like:
`make DEV=whatever`

File: readme.md [hunk 2]
```diff
@@ -4,14 +4,75 @@ Code repository for the Goose VI general platform board for the electrical archi
...
-cd /path/to/G6-MCU-Platform-Board
-make
+# this is an example, can replace DEV with whatever device you need
+bear -- make DEV=STM32G473CBTx
 ```
+
+### Code
+
+Write your code in the `platform` directory.
+
```
Comment (Quinn1876) [inline readme.md:53]: So the goal for this is to be common drivers and the "application code" will be for board specific code? For example, BMS code would be application code? but can driver would be platform code?

File: STM32G473CBTx/makefile [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+SZ = $(GCC_PATH)/$(PREFIX)size
+AR = $(GCC_PATH)/$(PREFIX)ar
+CPP_CC = $(GCC_PATH)/$(PREFIX)g++ -std=c++11
+else
+CC = $(PREFIX)gcc
+AS = $(PREFIX)gcc -x assembler-with-cpp
+CP = $(PREFIX)objcopy
+SZ = $(PREFIX)size
+AR = $(PREFIX)ar
+CPP_CC = $(PREFIX)g++ -std=c++11
```
Comment (Quinn1876) [inline STM32G473CBTx/makefile:75]: Is there a reason that we are only tagetting c++11?

File: readme.md (hunk 2 above)
Comment (RyanJMah) [reply readme.md:53]: Not exactly. I guess, all the code that is common to the MCU board would be in the platform layer (e.g., CAN, EEPROM, etc).

The makefile in this repo will build a static library `build/platform/platform.a` which can be linked with the application code (e.g., BMS, etc).

File: STM32G473CBTx/makefile (hunk 3 above)
Comment (RyanJMah) [reply STM32G473CBTx/makefile:75]: Not really. I know that C++ 20 depreceated compound assignment to voltaile (which is something we need), so I just picked something older than that.
- https://www.reddit.com/r/cpp/comments/jswz3z/compound_assignment_to_volatile_must_be/

If we ever need anything later than C++ 11 we can always just change that variable tho.

**Review by Nushaab-Syed — APPROVED** [review]

**Pull request merged**